### PR TITLE
Add new settings for the new health tab

### PR DESCRIPTION
**Pull request by msamper**
### Features and Changes

Adds three new settings behind a feature flag:

Organization Settings

- Run traffic query by default (Enabled by default)
- SRM Threshold (0.001 by default)

Data Source Exposure Query Settings

- Dimensions to use in traffic breakdown (empty by default) 

### Testing

- Make sure changes to the new organization settings are persisted
- Make sure the value for `Dimensions to use in traffic breakdown` is changed when on of its values is removed from `Dimensions` (i.e. country is one of the traffic dimensions but its removed from dimensions so we'll want to remove it from traffic dimensions as well) 

### Screenshots

Screenshot 2023-11-14 at 11 29 52 AM

Screenshot 2023-11-14 at 11 30 05 AM

### To-Do

- [x] Need to hook up the SRM threshold setting to the SRM warning
- [x] Fix old references to `SRM_THRESHOLD` and have the srmThreshold setting either return the value in mongo or the default

### CODE REVIEW

File: packages/front-end/pages/settings/index.tsx [hunk 1]
```diff
@@ -892,6 +895,48 @@ const GeneralSettingsPage = (): React.ReactElement => {
...
+                          }}
+                        />
+                      </div>
+
+                      <Field
+                        label="SRM threshold"
+                        type="number"
+                        step="0.001"
+                        max="0.5"
+                        min="0.001"
```
Comment (msamper) [inline packages/front-end/pages/settings/index.tsx:923]: @lukesonnet What should the max and min value for the SRM threshold be? Also curious if we want to highlight and provide messaging for the SRM threshold when it exceeds a certain value like we do for the p-value threshold

File: packages/front-end/pages/settings/index.tsx [hunk 2]
```diff
@@ -223,6 +223,8 @@ export const supportedCurrencies = {
   ZWL: "Zimbabwe Dollar (ZWL)",
 };
 
+export const DEFAULT_SRM_THRESHOLD = 0.001;
```
Comment (msamper) [inline packages/front-end/pages/settings/index.tsx:226]: Considered adding this to `shared/constants` but my understanding is this value will only be used by the front-end so I place it here

File: packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx [hunk 3]
```diff
@@ -71,15 +81,20 @@ export const AddEditExperimentAssignmentQueryModal: FC<EditExperimentAssignmentQ
   const userEnteredQuery = form.watch("query");
   const userEnteredDimensions = form.watch("dimensions");
   const userEnteredHasNameCol = form.watch("hasNameCol");
+  const userEnteredDimsForTraffic = form.watch("dimensionsForTraffic");
 
   const handleSubmit = form.handleSubmit(async (value) => {
-    await onSave(value);
+    // Do a JIT migration if user doesn't supply any dimensions for traffic
```
Comment (lukesonnet) [inline packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx:87]: Sorry for he back and forth but IDK if we should do this if we're going to choose to update the defaults later (e.g. in a couple of months). So maybe instead just leave it undefined.

**Review by lukesonnet — APPROVED** [review]
LGTM, minor nit and unsure about value of JIT now. We can add it later, I think.

File: packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx (hunk 3 above)
Comment (msamper) [reply packages/front-end/components/Settings/EditDataSource/ExperimentAssignmentQueries/AddEditExperimentAssignmentQueryModal.tsx:87]: Will do

**Pull request merged**